nymms/api/routes.py

```diff
@@ -0,0 +1,142 @@
+import arrow
+
+from flask import request
+from flask.ext.api import FlaskAPI
+from flask.ext.api import status
+
+from schematics.exceptions import ValidationError
+
+from nymms.state import sdb_state
+from nymms.suppress import sdb_suppress
+from nymms.schemas import APIResult, APISuppression, APIStateRecord
+from nymms.config import config
+from nymms.utils import aws_helper
+from nymms import schemas
+
+
+nymms_api = FlaskAPI(__name__)
+
+
+@nymms_api.route('/state', methods=['GET'])
+def state():
+    """
+    List current states.
+    """
+    region = config.settings['region']
+    domain = config.settings['state_domain']
+
+    state = sdb_state.SDBStateBackend(region, domain)
+    states = state.get_all_states(
```

Comment:
So - while working on my own API I was using this, but it had an issue - especially if you wanted to do anything like pagination with results.

http://docs.aws.amazon.com/AmazonSimpleDB/latest/DeveloperGuide/UsingSelect.html

SDB will, at max return 2500 records (100 by default) and a max of 1mb of results.  it then provides a token that you can requery.

I started to work with this with this method (I ended up replacing get_all_states with it):

```
    def filter(self, filters=None, order_by=None, consistent_read=True,
               max_items=None, next_token=None):
        order_by = order_by or getattr(self.schema, 'default_sort_order')

        query = "select * from %s" % self.domain_name
        if filters:
            query += " where "
            query += ' and '.join(filters)

        # TODO: This is kind of a weak way of dealing with the fact that in
        # order to order by something it has to be specified in the where
        # field.
        if order_by and order_by in query:
            query += " order by `%s`" % order_by

        results = []

        if max_items:
            max_items = int(max_items)
            query += " limit %s" % max_items

        logger.debug("Executing query: %s", query)
        query_results = self.domain.select(query,
                                           consistent_read=consistent_read,
                                           max_items=max_items,
                                           next_token=next_token)
        for item in query_results:
            results.append(self.deserialize(item))

        _next_token = query_results.next_token
        if _next_token:
            _next_token = _next_token.replace('\n', '')
        return (results, _next_token)
```

How would you expect this to work in your API?  Just keep doing filter() calls till you reach the end of the list of states?